main/main.go

```diff
@@ -216,6 +225,33 @@ func main() {
 
 	// Create listeners for each of the subnets configured as a source
 	errGroup, ctx := errgroup.WithContext(context.Background())
+
+	if cfg.DeciderPort != nil {
+		port := strconv.FormatUint(uint64(*cfg.DeciderPort), 10)
+
+		host := cfg.DeciderHost
+		if len(host) == 0 {
+			host = "localhost"
+		}
+
+		grpcClient, err = grpc.NewClient(
+			strings.Join([]string{host, port}, ":"),
+			grpc.WithTransportCredentials(
+				insecure.NewCredentials(),
+			),
+		)
+		if err != nil {
+			logger.Fatal(
+				"Failed to instantiate decider client",
+				zap.Error(err),
+			)
+			panic(err)
+		}
+		runtime.SetFinalizer(grpcClient, func(c *grpc.ClientConn) { c.Close() })
+		grpcClient.WaitForStateChange(ctx, connectivity.Ready)
```

Comment:
I believe WaitForStateChange is currently marked as experimental:

https://pkg.go.dev/google.golang.org/grpc#ClientConn.WaitForStateChange

Not sure if this is an issue, just wanted to point it out.

Reply:
thanks for pointing that out, I removed it in https://github.com/ava-labs/awm-relayer/pull/344/commits/584556eccf9ea1c2fa23db5adffdb15b5d210fd4#diff-327181d0a8c5e6b164561d7910f4eeffd41442d55b2a2788fda2aa2692f17ec0